core/trino-main/src/main/java/io/trino/execution/querystats/QueryRuleStats.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution.querystats;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.airlift.stats.TimeDistribution;
+import io.airlift.stats.TimeDistribution.TimeDistributionSnapshot;
+
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class QueryRuleStats
+{
+    private final String rule;
+    private long invocations;
+    private long applications;
+    private final TimeDistribution time = new TimeDistribution(TimeUnit.MICROSECONDS);
+    private long totalTime;
```

Comment:
`time.getCount()` is the same (`totalTime`)

Reply:
from class `TimeDistribution`:
```
getAvg() {
        return this.convertToUnit(this.total.getCount()) / this.getCount();
    }

getCount() {
      return this.digest.getCount();
  }
```

So `getCount` returns the number of points added, `this.total.getCount()` as a sum of points is not exposed.